legend-pure-core/legend-pure-m3-core/src/test/java/org/finos/legend/pure/m3/tests/elements/property/AbstractTestDefaultValue.java

```diff
@@ -126,5 +127,24 @@ public void testDefaultValueWithQualifiedProperty()
                 + "}\n"
         );
         execute("test():Boolean[1]");
-     }
+    }
+
+    @Test
+    public void testDefaultValueAssociationFailsCompile()
+    {
+        try
+        {
+            compileTestSource(
+                    "defaultValueSource.pure",
+                     "Class test::A { i:Integer[1]; }\n" +
+                            "Class test::B { j:Integer[1]; }\n" +
+                            "Association test::AB { a : test::A[1]; b : test::B[1] = ^test::B(j=2);}"
+            );
+        }
+        catch (RuntimeException e)
+        {
+            assertPureException(PureCompilationException.class, "Association properties should not have default values defined. Error in property: b", "defaultValueSource.pure", 3, 19, 3, 71, e);
+        }
```

Comment:
As this test stands, it will succeed if an exception is NOT thrown. Let's use Assert.assertThrows here.
```suggestion
        PureCompilationException e = Assert.assertThrows(PureCompilationException.class, () -> compileTestSource(
                    "defaultValueSource.pure",
                     "Class test::A { i:Integer[1]; }\n" +
                            "Class test::B { j:Integer[1]; }\n" +
                            "Association test::AB { a : test::A[1]; b : test::B[1] = ^test::B(j=2);}"
            ));
        assertPureException(PureCompilationException.class, "Association properties should not have default values defined. Error in property: b", "defaultValueSource.pure", 3, 19, 3, 71, e);
```